src/Symfony/Component/Serializer/Serializer.php

```diff
@@ -84,10 +84,12 @@ class Serializer implements SerializerInterface, ContextAwareNormalizerInterface
     /**
      * @param array<NormalizerInterface|DenormalizerInterface> $normalizers
      * @param array<EncoderInterface|DecoderInterface>         $encoders
+     * @param array<string, mixed>                             $defaultContext
      */
     public function __construct(
         private array $normalizers = [],
         array $encoders = [],
+        private array $defaultContext = [],
```

Comment:
Adding a new optional parameter to the constructor is expanding the public API. This can only happen as part of a new minor version as per semver. This should target 7.3 I think, even though it means not fixing this bug on lower branches.

Reply:
And then we need an entry in the Serializer's CHANGELOG

Reply:
And then you can update back the changes on `serializer.php` so that the service definition handles the default context.